plugin/plugin_settings.py

```diff
@@ -139,35 +118,35 @@ def load_settings(self):
         if 'project_base_name' in variables:
             self.project_base_name = variables['project_base_name']
 
-    def get_project_clang_flags(self):
+        # override nessesary settings from projects
+        self.__update_settings_from_project_if_needed()
+
+    def __update_settings_from_project_if_needed(self):
         """Get clang flags for the current project
 
         Returns:
             list(str): flags for clang, None if no project found
         """
-        try:
-            project_data = sublime.active_window().project_data()
-            log.debug(" project data: %s", project_data)
-            project_settings = project_data["settings"]
-            log.debug(" project settings: %s", project_settings)
-            project_flags = []
-            for flag in project_settings["clang_flags"]:
-                if flag.startswith('-I'):
-                    project_flags.append(self.__expand_include(flag))
-                elif flag.startswith('-std'):
-                    if 'c++' in flag:
-                        self.std_flag_cpp = flag
-                    else:
-                        self.std_flag_c = flag
-                else:
-                    # we just append everything else
-                    project_flags.append(flag)
-            log.debug(" project_flags: %s", project_flags)
-            return project_flags
-        except Exception as e:
-            log.error(" failed to read clang flags from project settings.")
-            log.error(" error is: %s.", e)
-            return None
+        log.debug(" overriding settings by project ones if needed:")
+        settings_handle = sublime.active_window().active_view().settings()
+        self.__load_vars_from_settings(settings_handle)
+        log.debug(" done.")
+
+    def __load_vars_from_settings(self, settings_handle):
+        for key, value in SettingsEnum.__dict__.items():
+            if key.startswith('__') or callable(key):
```

Comment:
Would maybe make sense to just use normal dict and iterate it using normal loop. Then you don't have to care about any existing properties.

Define dict like a static global:

SETTINGS_DICT = {...}